diff --git a/nets/facenet.py b/nets/facenet.py
index e7a6fcd..2ed9f89 100644
--- a/nets/facenet.py
+++ b/nets/facenet.py
@@ -64,7 +64,8 @@ class Facenet(nn.Module):
         self.Dropout    = nn.Dropout(1 - dropout_keep_prob)
         self.Bottleneck = nn.Linear(flat_shape, embedding_size,bias=False)
         self.last_bn    = nn.BatchNorm1d(embedding_size, eps=0.001, momentum=0.1, affine=True)
-        if mode == "train":
+        self.mode = mode
+        if self.mode == "train":
             self.classifier = nn.Linear(embedding_size, num_classes)
 
     def forward(self, x, mode = "predict"):
@@ -75,7 +76,10 @@ class Facenet(nn.Module):
             x = self.Dropout(x)
             x = self.Bottleneck(x)
             x = self.last_bn(x)
-            x = F.normalize(x, p=2, dim=1)
+            if self.mode == 'export':
+                return x
+            else:
+                x = F.normalize(x, p=2, dim=1)
             return x
         x = self.backbone(x)
         x = self.avg(x)
